### ISSUE TITLE
Workflows

### ISSUE THREAD

**Issue by shaw2thefloor** [CyVerseMetadataHack/HackathonIdeas#4]
One thing we have talked about since the start of COPO is the ability to have analysis pipelines stored within COPO profiles, which will allow experiments to be rerun easily. Can we explore how this would work? I understand the some sort of textual representation of a CV workflow can be exported. Can we store this alongside the data, and have a one click mechanism for arbitrary users to rerun the experiment?

**Comment by ramonawalls** [comment]
CyVerse stores all analyses, so they can be rerun. We should be able to export that as JSON or something to send to COPO. For workflows (a series of apps), I don't know how those are saved, but I suspect they are. This is also very similar to what is possible in SciApps (http://www.sciapps.org/).

**Comment by ramonawalls** [comment]
We also want to add more complex workflow/provenance tracking in CyVerse for managing data, not just for rerunning analyses.

**Comment by shaw2thefloor** [comment]
Indeed. From our perspective, we though it would be cool if users could see a useful CyVerse workflow which was stored in COPO, then run it on their own data. How feasible is this?

**Comment by froggleston** [comment]
There are already solutions for describing workflows such as CWL (Common Workflow Language) and YAML, so having some import/export functionality to help users package up workflows into submission pipelines would be an interesting idea.

**Comment by ramonawalls** [comment]
I've been reading about workflow-centric research objects today (e.g., https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4177597/, http://wf4ever.github.io/ro-primer/) which I now want for CyVerse. Seems very timely to this discussion.

**Comment by froggleston** [comment]
Sounds good. I know one of the authors, Stian, so we'd have a contact if required.
